py3status/modules/conky.py

```diff
@@ -0,0 +1,448 @@
+    {'full_text': '40K/116K (156K)', 'color': '#87cefa'},
+]
+
+processes
+[
+    {'full_text': 'Processes '}, {'full_text': u'342/0', 'color': '#00ffff'}
+]
+
+top
+[
+    {'full_text': 'Top '},
+    {'full_text': 'firefox-esr ', 'color': '#a9a9a9'},
+    {'full_text': '512M ', 'color': '#00bfff'},
+    {'full_text': '0.25% ', 'color': '#87cefa'},
+    {'full_text': 'htop ', 'color': '#a9a9a9'},
+    {'full_text': '2.93M ', 'color': '#00bfff'},
+    {'full_text': '0.17%', 'color': '#87cefa'},
+]
+
+top_mem
+[
+    {'full_text': 'Top Mem '},
+    {'full_text': 'chrome ', 'color': '#a9a9a9'},
+    {'full_text': '607M ', 'color': '#006400'},
+    {'full_text': '7.86% ', 'color': '#90ee90'},
+    {'full_text': 'thunderbird ', 'color': '#a9a9a9'},
+    {'full_text': '449M ', 'color': '#006400'},
+    {'full_text': '5.82%', 'color': '#90ee90'},
+]
+
+network
+[
+    {'full_text': 'Speed '},
+    {'color': '#ff6699', 'full_text': '15B/84B '},
+    {'full_text': 'Total '},
+    {'color': '#ff6699', 'full_text': '249MiB/4.27GiB'},
+]
+
+memory
+[
+    {'full_text': 'Memory '},
+    {'full_text': '2.68G/7.72G ', 'color': '#87cefa'},
+    {'full_text': '34% [###.......]', 'color': '#8ae234'}
+]
+
+swap
+[
+    {'full_text': 'Swap '},
+    {'full_text': '4.5MiB/7.72GiB ', 'color': '#f08080'},
+    {'full_text': '0% [..........]', 'color': '#a9a9a9'}
+]
+
+disk
+[
+    {'full_text': 'Home '},
+    {'full_text': '167G/431G ', 'color': '#ee82ee'},
+    {'full_text': '38% [####......]', 'color': '#8ae234'},
+]
+
+nvidia
+[
+    {'full_text': 'GPU Temp '}, {'full_text': '64 ', 'color': '#adff2f'},
+    {'full_text': 'GPU Freq '}, {'full_text': '460 ', 'color': '#adff2f'},
+    {'full_text': 'Mem Freq '}, {'full_text': '695', 'color': '#adff2f'},
+]
+
+nvidia
+[
+    {'full_text': 'MTR Freq '}, {'full_text': '1390 ', 'color': '#adff2f'},
+    {'full_text': 'Perf '}, {'full_text': '1 ', 'color': '#adff2f'},
+    {'full_text': 'Mem Perc '}, {'full_text': '61', 'color': '#adff2f'},
+]
+
+bar
+[
+    {'full_text': '#.... ', 'color': '#ffffff'},
+    {'full_text': '##... ', 'color': '#00FF00'},
+    {'full_text': '###.. ', 'color': '#FFA500'},
+    {'full_text': '####. ', 'color': '#FFFF00'},
+    {'full_text': '#####', 'color': '#FF0000'},
+]
+"""
+
+from subprocess import Popen, PIPE, STDOUT
+from threading import Thread
+from tempfile import NamedTemporaryFile
+from json import dumps
+from os import remove as os_remove
+
+STRING_NOT_INSTALLED = "not installed"
+STRING_MISSING_FORMAT = "missing format"
+
+
+class Py3status:
+    """
+    """
+
+    # available configuration parameters
+    config = {}
+    format = None
+    thresholds = []
+
+    def post_config_hook(self):
+        if not self.py3.check_commands("conky"):
+            raise Exception(STRING_NOT_INSTALLED)
+        elif not self.format:
+            raise Exception(STRING_MISSING_FORMAT)
+
+        # placeholders
+        self.delimiter = getattr(self, "delimiter", ".")
```

Comment:
`delimiter` is a hidden config? Do you expect this to be configurable? If not, let's hardcode `.`? If yes, how is user supposed to know about this, where is it documented?

Reply:
Yes on undocumented hidden trivial config. It's there just in case we need to troubleshoot something. This shouldn't be hardcoded.

Reply:
I don't feel it's the right approach to put such things in master, to ship software with debug blocks. If you ever need to trouble-shoot something, you can make modifications to the module itself...

Reply:
I think it would be more hassle to debug if we take them out, but sure, I can take them out.